src/tactic/ui/table/task_element_wdg.py

```diff
@@ -550,15 +557,91 @@ def preprocess(my):
 
 
         pipeline_codes = SObject.get_values(my.sobjects, 'pipeline_code', unique=True, no_exception=True)
-        pipelines = Search.eval("@SOBJECT(sthpw/pipeline['code','in','%s'])" %'|'.join(pipeline_codes))
+
+        # pipeline_codes can have an item that is something like $PROJECT/shot.
+        # when this happens, in the eval, $PROJECT is replaced by something like 'vfx'
+        # because there are quotes around it, it messes up the expression 
+        # So, replace $PROJECT here, and thus, getting rid of the code
+        project_code = Project.get_project_code()
+        if pipeline_codes:
+            pipeline_codes[0] = pipeline_codes[0].replace("$PROJECT", project_code)
+        pipelines = Search.eval("@SOBJECT(sthpw/pipeline['code','in','%s'])" % '|'.join(pipeline_codes) )
+
+        # get all of the processes that appear in all the pipelines, without duplicates
+        my.all_processes_set = set()
+        # ^ that's what the pipeline column should display (each process in the pipeline)
+
+        # the default pipeline is the longest pipeline in the table
+        # this is so that the processes appear in the right order.
+        # all other processes should follow after this
+        my.default_pipeline_list = []
+
+        # get all the processes
         if pipelines:
             for pipeline in pipelines:
                 processes = pipeline.get_processes()
+                # if this pipeline has more processes than the default, make this the default
+                if len(processes) > len(my.default_pipeline_list):
+                    my.default_pipeline_list = processes
                 pipeline_code = pipeline.get_code()
                 my.label_dict[pipeline_code] = {}
                 for process in processes:
+                    # put the processes found into a set to avoid duplicates.
+                    my.all_processes_set.add(process.get_name())
                     process_dict = my.label_dict.get(pipeline_code)
-                    process_dict[process.get_name()] = process.get_label()
+                    process_dict[process.get_name()] = process.get_label() 
+
+            # sort the processes, so that the task appears in the right order
+            my.default_pipeline_processes = []
+
+            # get an list of all the processes, in order
+            for process in my.default_pipeline_list:
+                my.default_pipeline_processes.append(process.get_name())
```

Comment:
my.default_pipeline_processes = [x.get_name() for x in my.default_pipeline_list]